dockerfile_functions.sh

```diff
@@ -76,32 +76,25 @@ EOI
 # Select the alpine OS packages.
 # Install GNU glibc as this OpenJDK build is compiled against glibc and not musl.
 print_alpine_pkg() {
-	cat >> $1 <<'EOI'
+	cat >> "$1" <<EOI
+COPY --from=$target_registry/openjdk-glibc-builder:2.28-$(uname -m) /usr/glibc-compat/. /usr/glibc-compat/
+COPY --from=$target_registry/openjdk-gcc-builder:8.2.0-$(uname -m) /usr/glibc-compat/. /usr/glibc-compat/lib/
+COPY --from=$target_registry/openjdk-zlib-builder:1.2.11-$(uname -m) /usr/glibc-compat/. /usr/glibc-compat/lib/
```

Comment:
In general - approach  to separate different dependencies is good. But actually images on which we rely on must be improved:
```
REPOSITORY                          TAG                            IMAGE ID            CREATED             SIZE
adoptopenjdk/openjdk-zlib-builder   1.2.11-x86_64                  afffe56287a2        7 minutes ago       315MB
```
315 MB for one dependency is huge overhead ! And if I look what we actually want to copy there: 
* `docker run -it adoptopenjdk/openjdk-zlib-builder:1.2.11-x86_64 bash`
* `cd /usr/glibc-compat/`
* `du -h`

gives :
`296K    .` - so  in order to copy 400 kb we need to pull image having 325 MB !
And this is for all three of them (glibc - 669 MB to get 142 MB, gcc - 2 GB ).

Solution ? Use multi-stage builds in your repos - https://docs.docker.com/develop/develop-images/multistage-build/

Just change :
```
FROM ubuntu:18.04
....
```
to 
```
FROM ubuntu:18.04 as builder
....
rest of the Dockerfile
....
FROM scratch

COPY --from=builder /usr/glibc-compat /usr/glibc-compat
```
it will produce images containing exactly files we would like to copy, which are significantly smaller.

Reply:
Additional remark - previously we used binaries (https://archive.archlinux.org/packages/g/gcc-libs/gcc-libs-8.2.1%2B20180831-1-x86_64.pkg.tar.xz), currently you build all dependencies from source (in your images) - are they 100% the same ?